components/webgpu/gpu_error.rs

```diff
@@ -0,0 +1,101 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+//! Error scopes and GPUError types
+
+use std::fmt;
+
+use serde::{Deserialize, Serialize};
+
+use crate::wgc;
+
+/// <https://www.w3.org/TR/webgpu/#gpu-error-scope>
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+pub(crate) struct ErrorScope {
+    // we only store first error
+    pub errors: Option<Error>,
```

Comment:
Should we not make this a list, and just return "any" when popped, with "any" being just the first one for now?

Reply:
That would be per spec, but why would we store multiple errors if we only use one?

Reply:
Because per the spec you should store them all but only use one anyway, the only difference is how that one is chosen. In the current implementation we could do something simple like always choosing the  last or the first, and add a comment about this at the point where this is done.

Reply:
I will change to vec, but generally the first error is the right one, although spec is very vague about that:
> Let error be any one of the items in scope.[[[errors]]](https://www.w3.org/TR/webgpu/#dom-gpu-error-scope-errors-slot)

last one can be from previous operation failing:
> For any two errors E1 and E2 in the list, if E2 was caused by E1, E2 should not be the one selected.